src/machines/builder_route/selection.config.js

```diff
@@ -0,0 +1,130 @@
+export const selectionConfig = {
+	initial: "CheckingSelection",
+	on: {
+		SELECTION_CHANGED: {
+			target: "Selecting.CheckingSelection",
+			actions: ['setURL']
+		},
+		SELECTION_RESET: {
+			target: "Selecting.SelectionIncomplete",
+			actions: [
+				'deleteNestedAggs',
+				'setURL'
+			]
+		}
+	},
+	states: {
+		CheckingSelection: {
+			on: {
+				'': [
+					{
+						target: 'SelectionComplete',
+						cond: 'isSelectionComplete',
+						actions: ['spawnNestedAgg']
+					},
+					{ target: 'SelectionIncomplete' }
+				]
+			}
+		},
+		SelectionIncomplete: {},
+		SelectionComplete: {
+			initial: "CheckingQuery",
+			on: {
+				QUERY_CHANGED: {
+					target: "SelectionComplete.CheckingQuery",
+					actions: ['setURL']
+				}
+			},
+			states: {
+				CheckingQuery: {
+					id: "CheckingQuery",
+					on: {
+						'': [
+							{
+								target: 'QueryReady',
+								cond: 'isQueryReady'
+							},
+							{ target: 'QueryNotReady' }
+						]
+					}
+				},
+				QueryNotReady: {
+					id: "QueryNotReady",
+				},
+				QueryReady: {
+					id: "QueryReady",
+					initial: "CheckMatching",
+					states: {
+						CheckMatching: {
+							onEntry: ["verifyMatching"],
```

Comment:
I think `verifyMatching` isn't needed, as it should be implicit in the guard `isMatching`